daphne/src/hpke.rs

```diff
@@ -181,9 +177,23 @@ impl HpkeReceiverConfig {
         }
     }
 
-    /// Create a new HPKE receiver context given an HpkeConfig and a corresponding secret key.
-    pub fn new(config: HpkeConfig, secret_key: Vec<u8>) -> Self {
-        HpkeReceiverConfig { config, secret_key }
+    /// Create a new HPKE receiver context given an HpkeConfig and a corresponding private key.
+    /// Returns an error if the public key does not correspond to the private_key.
+    pub fn try_from(config: HpkeConfig, private_key: HpkePrivateKey) -> Result<Self, DapError> {
```

Comment:
readability: Rust best practice is to not overload the names of common trait methods (i.e., TryFrom::try_from() in this case). Could we call this something like try_from_parts(), or maybe try_from_config_and_key()? Alternatively, we could implement `TryFrom<(HpkeConfig, HpkePrivateKey)> for `HpkeReceiverConfig`.

Reply:
done